src/BlueprintParser.h

```diff
@@ -357,14 +357,37 @@ namespace snowcrash {
                 return;
             }
 
-            // Otherwise, add the respective entries to the tables
-            if (typeDefinition.typeSpecification.name.base != mson::UndefinedTypeName) {
+            mson::BaseTypeName baseTypeName = typeDefinition.typeSpecification.name.base;
+
+            // Initialize an entry in the dependency table if not found
+            if (pd.namedTypeDependencyTable.find(identifier) != pd.namedTypeDependencyTable.end()) {
```

Comment:
Better design: in a future this really needs to be split into multiple functions – this one is already 200 lines long and it is quite hard to follow what is happening here.